woocommerce/woocommerce-android
# Dark Mode: Settings Option

Fixes #1583 by adding a new setting option in the app to change the theme. This PR covers the following:
- Adds the new "Appearance" settings option.
- Clicking this option opens a dialog to choose a theme.
- Selecting a theme applies that theme to the app. 
- Selected theme is saved to SharedPrefs and loaded when the app starts.
- Theme tweaks to properly theme the "Appearance" chooser dialog.
Does not include
- The track event: I've submitted a question to the meeting channel to determine if this setting option needs to be tracked. If it does, I'll open a ticket and have it added in a different PR. 

Note: This PR includes some tweaks to the main themes. You'll notice that the `primaryColor` and `secondaryColor` have been swapped in the main theme. This is done so the desired style of the dialog follows the current dark/light mode designs where the radio buttons are purple and the borderless button is pink. See the comments in #1583 for additional information. 

The options available in the "Appearance" chooser dialog are different depending on the version of Android:

![api-29](https://user-images.githubusercontent.com/5810477/71392248-a88f4200-25c4-11ea-8427-354d5e1af510.png)

<img width="664" alt="api-28-full" src="https://user-images.githubusercontent.com/5810477/71392271-b775f480-25c4-11ea-8286-1312bf8b46a1.png">

## To Test

On a device running API 29 in light mode:
1. Open Settings
2. Verify the "Appearance" settings option is displayed and the current setting is "System default".
3. Click the "Appearance" settings option. 
4. Verify the chooser is displayed and matches the design. 
5. Select "dark". The dialog should immediately close, the appbar and system navigation bar should turn black. 
6. Click the "Appearance" settings option again. Verify the dialog is now using the dark theme and matches the design. 
7. Select "light". Verify the appbar and system navigation bar both turned back to light theme. 
8. Click the "Appearance" settings option and tap "System default". 
9. Go into the device settings and change to use the Dark theme. 
10. Go back to the app and verify the app has changed over to using the dark theme (system navigation bar and appbar should now be black). 
11. Turn off dark mode - verify app returns to light mode. 

On a device running API < 29 in light mode:
1. Open Settings
2. Verify the "Appearance" settings option is displayed and the current setting is "Set by battery saver".
3. Click the "Appearance" settings option. 
4. Verify the chooser is displayed and matches the design. 
5. Select "dark". The dialog should immediately close and the appbar and system navigation bar should turn black. 
6. Click the "Appearance" settings option again. Verify the dialog is now using the dark theme and matches the design. 
7. Select "light". Verify the appbar and system navigation bar both turned back to light theme. 
8. Click the "Appearance" settings option and tap "Set by battery saver". 
9. Pull down the notification shade on the device and toggle "Battery Saver" on.
10. Verify the app has changed over to using the dark theme (system navigation bar and appbar should now be black). 
11. Turn off "battery saver", verify app returns to light mode.
